server/config.go

```diff
@@ -0,0 +1,25 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package server
+
+import (
+	"errors"
+	"time"
+)
+
+type ShipperServerConfig struct {
+	// PollingInterval is an interval for polling queue metrics and updating the indices.
+	// Must be greater than 0
+	PollingInterval time.Duration
```

Comment:
Shouldn't we have a default polling interval?

Reply:
Also since this controls the persisted index polling interval it is going to interact with filebeat's `registry.flush` configuration at least: https://www.elastic.co/guide/en/beats/filebeat/current/configuration-general-options.html#_registry_flush

Setting `registry.flush` to a value < the shipper polling interval won't have any effect with this implementation. There are use cases out there with the filebeat `registry.flush` set to 0.

Reply:
If there is a better way to get a real time notification without polling, I'm naturally up for it. I used what was available to me.

Reply:
> Shouldn't we have a default polling interval?

Currently we don't have any proper configuration in the shipper. It should be addressed in a wider way setting up the whole configuration system like we do in Beats.

Reply:
> so that updates can be observed as fast as they occur?

It's possible but I'm not sure it's desirable -- it would make for a potentially very noisy channel (in particular even low-volume inputs would still get notified about every round-trip for _all_ data) and require a more complex queue API, and i don't see much payoff for the expected use cases. We already can't provide perfect granularity simply because we don't guarantee the output workers preserve the ordering of events, so it's a best-effort watermark not a real-time invariant.

(In fact, thinking through that tradeoff has led me to more drastic suggestions, see next comment)